Doc/library/ast.rst

```diff
@@ -917,6 +917,25 @@ Statements
             type_ignores=[])
 
 
+.. class:: TypeAlias(name, typeparams, value)
+
+   A type alias created through the :keyword:`type` statement. ``name``
+   is the name of the alias, ``typeparams`` is a list of
+   :ref:`type parameters <ast-type-params>`, and ``value`` is the value of the
+   type alias.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('type Alias = int'), indent=4))
+        Module(
+            body=[
+                TypeAlias(
+                    name=Name(id='Alias', ctx=Store()),
+                    typeparams=[],
```

Comment:
Is it too late to change this attribute so that it's `type_params`? Nearly all other attributes on AST nodes in Python use snake_case -- `type_ignores`, `decorator_list`, `end_lineno`, `type_comment`, `end_col_offset`, `format_spec`, `is_async`, `kw_defaults`, `context_expr`, `optional_vars`, `kwd_attrs`, `kwd_patterns`. And the attribute on `TypeAlias`es, functions and classes at runtime is `__type_params__`, not `__typeparams__`.

There's a few examples of smushedtogethercase attributes on AST nodes -- `finalybody`, `kwonlyargs`, `vararg`, `asname`. But they're definitely in the minority.

Reply:
It is explicitly specified this way in the PEP: https://peps.python.org/pep-0695/#ast-changes. But I agree that `type_params` would be cleaner.

Reply:
Marking this as resolved for now, will update this pending the resolution of #104656.

Reply:
Need to update this now ;)

Reply:
Find-replaced it across the file